core/src/main/java/org/arquillian/smart/testing/scm/git/GitChangeResolver.java

```diff
@@ -59,23 +59,35 @@ public GitChangeResolver(File projectDir, Scm scm) {
     public GitChangeResolver(File dir, String previous, String head) {
         this.previous = previous;
         this.head = head;
-        final FileRepositoryBuilder builder = new FileRepositoryBuilder();
-        try {
-            this.git = new Git(builder.readEnvironment().findGitDir(dir).build());
-        } catch (IOException e) {
-            throw new IllegalArgumentException("Unable to find git repository for path " + dir.getAbsolutePath(), e);
+        final FileRepositoryBuilder fileRepositoryBuilder = getFileRepositoryBuilder(dir);
+        if (fileRepositoryBuilder.getGitDir() != null) {
+            try {
+                this.git = new Git(fileRepositoryBuilder.build());
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+            this.repoRoot = git.getRepository().getDirectory().getParentFile();
+        } else {
+            logger.warn(String.format("Unable to find a git repository for the path %s - some strategies won't work properly."
+                + " Make sure it is a git repository.", dir.getAbsolutePath()));
         }
-        this.repoRoot = git.getRepository().getDirectory().getParentFile();
     }
 
     @Override
     public void close() throws Exception {
-        git.close();
+        if (git != null) {
+            git.close();
+        }
     }
 
     @Override
-    public Set<Change> diff() {
-        final Set<Change> allChanges= new HashSet<>();
+    public Set<Change> diff(String strategy) {
+        final Set<Change> allChanges = new HashSet<>();
+        if (!strategy.isEmpty() && git == null) {
```

Comment:
In case the strategy is `null`, then you get NPE.
In addition, swap the conditions - it makes more sense to check if the git is null as the first condition

Reply:
Here I assumed strategy as `""` not null. that's why I am checking `isEmpty()`

Reply:
see https://github.com/arquillian/smart-testing/pull/221#discussion_r144244237